lifx_client.py

```
@@ -98,26 +100,37 @@ def __repr__(self):
 
         self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self.sock.settimeout(0.5)
-        self.sock.bind((bind_ip, 0))
-        self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-
-        # Start listener thread
+        try:
+            self.sock.bind((bind_ip, 0))
+        except OSError as e:
+            print(f"Warning: Could not bind to {bind_ip}: {e}")
+            self.sock.bind(("0.0.0.0", 0))
```

Ruff (0.15.12)

[error] Possible binding to all interfaces

(S104)